test/expand_test.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+import {expand} from '../src/expand.js';
+import {Tensor} from '../src/lib/tensor.js';
+import * as utils from './utils.js';
+
+describe('test expand', function() {
+  function testExpand(input, newShape, expected) {
+    const tensor = new Tensor(input.shape, input.data);
+    const outputTensor = expand(tensor, newShape);
+    utils.checkShape(outputTensor, expected.shape);
+    utils.checkValue(outputTensor, expected.data);
+  }
+
+  it('expand changed dimensions', function() {
```

Comment:
Please add a more test for testing 0D expand to ND, you can refer to https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/dml/graph_impl_test.cc;l=1456, thanks.

Reply:
I included some cases [here](https://github.com/webmachinelearning/webnn/pull/478#discussion_r1396805359), where new shape is longer and shorter than input.rank (which is perfectly fine, so long as they are broadcast compatible with each other):

input.shape = [2,1,4], newShape = [3,1], output.shape = [2,3,4]
input.shape = [2,1,4], newShape = [5,1,3,4], output.shape = [5,2,3,4]
input.shape = [], newShape = [2,3], output.shape = [2,3]
input.shape = [2,3], newShape = [], output.shape = [2,3]

Reply:
I have added these cases. PTAL:)